13_07_2017/3D_slab_modes.py

```diff
@@ -0,0 +1,779 @@
+        mod = int(4 * nx / 100)
+        mod_y = int(np.ceil(mod / y_spacing))
+        
+        # Get the data xi=displacement, v=velocity, b=mag field
+        if show_disp_top == True or show_disp_front == True:
+            xixvals = np.real(np.repeat(sf.xix(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2))
+            xizvals = np.real(np.repeat(sf.xiz(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2))
+            xiyvals = np.real(np.repeat(sf.xiy(mode, xvals, zvals, t, W, K)[:, :, np.newaxis], ny, axis=2))
+        
+        if show_vel_front == True or show_vel_top == True:
+            vxvals = np.real(np.repeat(sf.vx(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2))
+            vzvals = np.real(np.repeat(sf.vz(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2))
+            vyvals = np.real(np.repeat(sf.vy(mode, xvals, zvals, t, K)[:, :, np.newaxis], ny, axis=2))
+        
+        if show_vel_front_pert == True or show_vel_top_pert == True:
+            vxvals = np.real(np.repeat(sf.vx_pert(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2))
+            vzvals = np.real(np.repeat(sf.vz_pert(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2))
+            vyvals = np.zeros_like(vxvals)
+        
+        # Axis is defined on the mag field so we have to set up this data
+        bxvals = np.real(np.repeat(sf.bx(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2))
+        byvals = np.real(np.repeat(sf.by(mode, xvals, zvals, t, K)[:, :, np.newaxis], ny, axis=2))
+        bz_eq3d = np.repeat(sf.bz_eq(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2)
+        bzvals = np.real(np.repeat(-sf.bz(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2) +
+                         bz_eq3d)
+        
+        # displacement at the right and left boundaries
+        if show_boundary == True:
+            xix_boundary_r_vals = np.real(np.repeat(K + sf.xix_boundary(mode, zvals, t, W, K, R1, boundary='r')[:, np.newaxis], ny, axis=1))
+            xix_boundary_l_vals = np.real(np.repeat(-K + sf.xix_boundary(mode, zvals, t, W, K, R1, boundary='l')[:, np.newaxis], ny, axis=1))
+        
+        if show_density == True:
+            rho_vals = np.real(np.repeat(sf.rho(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2))
+        
+        if show_density_pert == True:
+            rho_vals = np.real(np.repeat(sf.rho_pert(mode, xvals, zvals, t, W, K, R1)[:, :, np.newaxis], ny, axis=2))
+        
+        bxvals_t = bxvals
+        byvals_t = byvals
+        bzvals_t = bzvals
+        
+        if show_disp_top == True or show_disp_front == True:
+            xixvals_t = xixvals
+            xiyvals_t = xiyvals
+            xizvals_t = xizvals
+            
+        if np.array([show_vel_top, show_vel_top_pert, show_vel_front, show_vel_front_pert]).any() == True:
```

Comment:
These variables are all just single `True` or `False` values right? If so then `if True in [show_vel_top, etc, etc]:` will do the same thing and look a bit nicer.